core/src/main/scala/kafka/server/DelayedFetch.scala

```diff
@@ -154,9 +154,9 @@ class DelayedFetch(
    * Upon completion, read whatever data is available and pass to the complete callback
    */
   override def onComplete(): Unit = {
-    val fetchInfos = fetchPartitionStatus.map { case (tp, status) =>
+    val fetchInfos = fetchPartitionStatus.asScala.map { case (tp, status) =>
       tp -> status.fetchInfo
-    }
+    }.toSeq
```

Comment:
Is it possible to have duplicate TPs in this "fetch" path?

Reply:
Also, we are doing two conversions here - you can avoid one by using a view or iterator after `asScala` and using `toBuffer` instead of `toSeq` (the latter can result in a lazy collection being created which can result in problems).